invenio_rdm_records/contrib/thesis/custom_fields.py

```diff
@@ -42,6 +44,8 @@ def mapping(self):
                 "university": {"type": "keyword"},
                 "department": {"type": "keyword"},
                 "type": {"type": "keyword"},
+                "date_submitted": {"type": "keyword"},
```

Comment:
I might be wrong here, but I believe we agreed during the workshop session that we would have a `dates` field to which we will initially add only `submitted` and `defended`, somewhat artificially using the interface.

Reply:
> I think the conclusion was that it would be too difficult to port the date functionality into custom fields. But if it's possible that would be a better option.

yes that was the conclusion, I followed the last comment added in the issue by Tom. We can always improve in the future versions - but at this moment I couldn't unfortunately dig deeper on this

Reply:
Another point of the discussion in the workshop, was that we could have `dates` (with the limited types), and just handle the complexity in the UI by just showing two fields for input that serialize to a list. This would leave open the door for other date types, without having to go through adding more mapping fields to the custom field.

The only clear benefit I see with having them separate is that you can directly do a search query on either type of date (which wouldn't be possible if it was a list of objects, because of the usual ElasticSearch array-of-object flatenning issues).

> Disclaimer: I have zero stakes on the table for adding a thesis field, so take all this with a grain of salt 😅